pandas/core/generic.py

```diff
@@ -5885,25 +5885,61 @@ def pipe(
         Notes
         -----
         Use ``.pipe`` when chaining together functions that expect
-        Series, DataFrames or GroupBy objects. Instead of writing
+        Series, DataFrames or GroupBy objects.
 
-        >>> func(g(h(df), arg1=a), arg2=b, arg3=c)  # doctest: +SKIP
+        Examples
+        --------
+        Constructing a income DataFrame from a dictionary.
+
+        >>> data = [[8000, 1000], [9500, np.nan], [5000, 2000]]
+        >>> income = pd.DataFrame(data, columns=['Salary', 'Others'])
+        >>> income
+           Salary  Others
+        0    8000  1000.0
+        1    9500     NaN
+        2    5000  2000.0
+
+        Functions that perform tax reductions on an income DataFrame.
+
+        >>> def federal_tax(income):
+        ...     return income * 0.9
+        >>> def state_tax(income, rate):
+        ...     return income * (1 - rate)
+        >>> def national_insurance(income, rate, rate_increase):
+        ...     new_rate = rate + rate_increase
+        ...     return income * (1 - new_rate)
+
+        Instead of writing
+
+        >>> (national_insurance(state_tax(federal_tax(income), rate=0.12),
+        ...                     rate=0.05,
+        ...                     rate_increase=0.02))  # doctest: +SKIP
 
         You can write
 
-        >>> (df.pipe(h)
-        ...    .pipe(g, arg1=a)
-        ...    .pipe(func, arg2=b, arg3=c)
-        ... )  # doctest: +SKIP
+        >>> (income.pipe(federal_tax)
+        ...        .pipe(state_tax, rate=0.12)
+        ...        .pipe(national_insurance, rate=0.05, rate_increase=0.02))
```

Comment:
Very nice example, so clear compared to the previos `g`, `h`...

I think it'd be even a bit more clear if you call those functions `subtract_federal_tax`, `substract_state_tax`... Just the name of the tax makes a bit less obvious what's going on in my opinion. What do you think?